### PR TITLE
Feature/frontend file upload UI

### PR DESCRIPTION
**Pull request by Akrya**
#### Review Checklist
- [x] The requirements on the description have been met
- [x] The relevant documentation has been updated:
    - [x] code comments for classes / methods / complex statements
    - [x] package readme
    - [x] documentation for high level approaches or product info
- [ ] Tests:
    - [ ] new tests written
    - [x] existing affected tests updated
    - [x] all green and passing
- [ ] Tested by reviewer

No interaction with backend yet

### CODE REVIEW

File: app/src/pages/MainPage.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    IonCardContent,
+    IonCardTitle,
+    IonInput
+  } from '@ionic/react';
+  import { add, arrowBack} from 'ionicons/icons';
+  import React, { useState } from 'react';
+  import { useParams } from 'react-router';
+  import './MainPage.css';
+    
+  interface Document {
```
Comment (Cvos0162) [inline app/src/pages/MainPage.tsx:21]: Not needed

File: app/src/pages/MainPage.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            "Wellington, the capital of New Zealand, sits near the North Island’s southernmost point on the Cook Strait.",
+    },
+    {
+        title: "Project2",
+        description:
+            "Auckland, based around 2 large harbours, is a major city in the north of New Zealand’s North Island.",
+    },
+];
+  
+  const ProjectPage: React.FC = () => {
```
Comment (Cvos0162) [inline app/src/pages/MainPage.tsx:63]: rename please. we got duplicate. gotta match to the filename

File: app/src/pages/MainPage.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        
+        </IonContent>
+
+        
+
+      </IonPage>
+    );
+  };
+  
+  export default ProjectPage;
```
Comment (Cvos0162) [inline app/src/pages/MainPage.tsx:112]: Same here

File: app/src/App.tsx [hunk 4]
```diff
@@ -32,6 +33,7 @@ const App: React.FC = () => {
           <IonRouterOutlet id="main">
             <Route path="/project/:name" component={ProjectPage} exact />
             <Route path="/auth" component={LoginPage} exact />
+            <Route path="/mainpage" component={MainPage} exact />
```
Comment (Cvos0162) [inline app/src/App.tsx:36]: maybe consider having root / as the main page?

Comment (Cvos0162) [reply app/src/App.tsx:36]: Just an idea

**Review by Cvos0162 — CHANGES_REQUESTED** [review]

**Comment by tnzhu** [comment]
Hey, overall the page looks great, but my understanding was that this page should allow users to select a file to upload. I don't see this functionality implemented here?

**Comment by Akrya** [comment]
> Hey, overall the page looks great, but my understanding was that this page should allow users to select a file to upload. I don't see this functionality implemented here?
I assumed the main page was for creating a project, and you can upload files inside the "project page".  There is no actual functionality yet but the UI has been set up if you go on the path  "/project/:name".

**Comment by tnzhu** [comment]
> > Hey, overall the page looks great, but my understanding was that this page should allow users to select a file to upload. I don't see this functionality implemented here?
> 
> I assumed the main page was for creating a project, and you can upload files inside the "project page". There is no actual functionality yet but the UI has been set up if you go on the path "/project/:name".

Ah, sweet, I didn't see that we had individual project pages. Looking good!

**Review by Cvos0162 — APPROVED** [review]
LGTM!

**Pull request merged**